src/libPMacc/include/mappings/threads/ThreadCollective.hpp

```diff
@@ -32,72 +32,61 @@
+    using DomainSize = typename T_DataDomain::FullSuperCellSize;
+    // offset (in elements per dimension) from the GUARD origin to the CORE
+    using OffsetOrigin = typename T_DataDomain::OffsetOrigin;
 
-    /** full size of the domain including the guards around SuperCellSize */
-    using FullSuperCellSize = typename T_BlockDomain::FullSuperCellSize;
-
-    /** size of the negative guard */
-    using OffsetOrigin = typename T_BlockDomain::OffsetOrigin;
-
-    /** number of worker performing the functior */
     static constexpr uint32_t numWorkers = T_numWorkers;
+    static constexpr uint32_t dim = T_DataDomain::Dim;
 
-    /** dimensionality of the index domain */
-    static constexpr uint32_t dim = T_BlockDomain::Dim;
-
-    /** index of the worker: range [0;numWorkers) */
-    PMACC_ALIGN( m_workerIdx, uint32_t const );
+    const PMACC_ALIGN(
+        m_workerIdx,
+        uint32_t
+    );
 
 public:
 
     /** constructor
      *
-     * @param workerIdx worker index
+     * @param workerIdx index of the worker
      */
     DINLINE ThreadCollective( uint32_t const workerIdx ) :
         m_workerIdx( workerIdx )
-    { }
-
-    /** constructor
-     *
-     * @warning this constructor is **deprecated** (assuming that the index domain
-     *          and the number of workers needs to be euqal is not good and avoid the
-     *          usage of alpaka)
-     *
-     * @param nDimWorkerIdx n dimensional worker index
-     */
-    DINLINE ThreadCollective( DataSpace< dim > const nDimWorkerIdx ) :
-        m_workerIdx( DataSpaceOperations< dim >::template map< SuperCellSize >( nDimWorkerIdx ) )
-    { }
+    {
+    }
 
-    /** execute a functor for each data point
+    /** execute the user functor for each element in the full domain
      *
-     * @tparam T_Functor type of the functor
-     * @tparam T_Args type of arguments given to the functor
+     * @tparam T_Functor type of the user functor, must have a `void operator()`
+     *                   with as many arguments as args contains
+     * @tapram  T_Args type of the arguments, each type must implement an operator
+     *                 `template<typename T, typnme R> R operator(T)`
      *
-     * @param functor functor which is called for each data element in the domain
-     *                spanned by T_BlockDomain
-     * @param args the result of each argument of the `operator( relativeNDimensionalIdx )` call
-     *             with the relative index inside `T_BlockDomain` based on
-     *             the origin `T_BlockDomain::OffsetOrigin` is passed to the functor
+     * @param functor user defined functor
+     * @param args arguments passed to the functor
+     *             The method `template<typename T, typnme R> R operator(T)`
+     *             is called for each argument, the result is passed to the
+     *             functor `functor::operator()`.
+     *             `T` is a N-dimensional vector of an index relative to the origin
```

Comment:
It is spoken "an EN-dimensional". Are you really saying "a Nnnnnn-dimensional"? Sounds funny. A bit Polish. They love consonants without vocals. "W" is a very often used word in Poland. Anyway, no real reason to block this PR...